drivers/watchdog/wdt_mchp_xec.c

```diff
@@ -0,0 +1,189 @@
+#include <assert.h>
+
+#define WDT_XEC_REG_BASE						\
+	((WDT_Type *)(DT_INST_0_MICROCHIP_XEC_WATCHDOG_BASE_ADDRESS))
+
+struct wdt_xec_data {
+	wdt_callback_t cb;
+	bool timeout_installed;
+};
+
+
+DEVICE_DECLARE(wdt_xec);
+
+static int wdt_xec_setup(struct device *dev, u8_t options)
+{
+	WDT_Type *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (wdt_regs->CTRL & MCHP_WDT_CTRL_EN) {
+		return -EBUSY;
+	}
+
+	if (!data->timeout_installed) {
+		LOG_ERR("No valid WDT timeout installed");
+		return -EINVAL;
+	}
+
+	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
+		LOG_WRN("WDT_OPT_PAUSE_IN_SLEEP is not supported");
+		return -ENOTSUP;
+	}
+
+	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
+		wdt_regs->CTRL |= MCHP_WDT_CTRL_JTAG_STALL_EN;
+	} else {
+		wdt_regs->CTRL &= ~MCHP_WDT_CTRL_JTAG_STALL_EN;
+	}
+
+	wdt_regs->CTRL |= MCHP_WDT_CTRL_EN;
+
+	LOG_DBG("WDT Setup and enabled");
+
+	return 0;
+}
+
+static int wdt_xec_disable(struct device *dev)
+{
+	WDT_Type *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (!(wdt_regs->CTRL & MCHP_WDT_CTRL_EN)) {
+		return -EALREADY;
+	}
+
+	wdt_regs->CTRL &= ~MCHP_WDT_CTRL_EN;
+	data->timeout_installed = false;
+
+	LOG_DBG("WDT Disabled");
+
+	return 0;
+}
+
+static int wdt_xec_install_timeout(struct device *dev,
+				   const struct wdt_timeout_cfg *config)
+{
+	WDT_Type *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (wdt_regs->CTRL & MCHP_WDT_CTRL_EN) {
+		return -EBUSY;
+	}
+
+	if (config->window.min > 0U) {
+		data->timeout_installed = false;
+		return -EINVAL;
+	}
+
+	wdt_regs->LOAD = 0;
+
+	data->cb = config->callback;
+	if (data->cb) {
+		wdt_regs->CTRL |= MCHP_WDT_CTRL_MODE_IRQ;
+		wdt_regs->IEN |= MCHP_WDT_IEN_EVENT_IRQ_EN;
+
+		LOG_DBG("WDT callback enabled");
+	} else {
+		/* Setting WDT_FLAG_RESET_SOC or not will have no effect:
+		 * even after the cb, if anything is done, SoC will reset
+		 */
+		wdt_regs->CTRL &= ~MCHP_WDT_CTRL_MODE_IRQ;
+		wdt_regs->IEN &= ~MCHP_WDT_IEN_EVENT_IRQ_EN;
+
+		LOG_DBG("WDT Reset enabled");
+	}
+
+	/* Since it almost takes 1ms to decrement the load register
+	 * (See datasheet 18.6.1.4: 33/32.768 KHz = 1.007ms)
+	 * Let's use the given window directly.
+	 */
+	wdt_regs->LOAD = config->window.max;
+
+	data->timeout_installed = true;
+
+	return 0;
+}
+
+static int wdt_xec_feed(struct device *dev, int channel_id)
+{
+	WDT_Type *wdt_regs = WDT_XEC_REG_BASE;
+
+	ARG_UNUSED(dev);
+	ARG_UNUSED(channel_id);
+
+	if (!(wdt_regs->CTRL & MCHP_WDT_CTRL_EN)) {
+		return -EINVAL;
+	}
+
+	LOG_DBG("WDT Kicking");
+
+	wdt_regs->KICK = 1;
+
+	return 0;
+}
+
+static void wdt_xec_isr(struct device *dev)
+{
+	WDT_Type *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (!(MCHP_GIRQ_RESULT(MCHP_WDT_GIRQ) & MCHP_WDT_GIRQ_VAL)) {
+		return;
+	}
+
+	MCHP_GIRQ_SRC(MCHP_WDT_GIRQ) &= ~MCHP_WDT_GIRQ_VAL;
+
+	LOG_DBG("WDT ISR");
+
+	if (data->cb != NULL) {
```

Comment:
Minor stuff: check the pointer in the same way you did in your counter driver. In this case should be if (data->cb)